terragraph_planner/common/rf/link_budget_calculator.py

```diff
@@ -582,25 +567,36 @@ def fspl_based_estimation(
         rx_el_deviation=rx_el_deviation,
     )
     np_dbm = get_noise_power(rx_sector_params)
-    link_budget = get_measurements_from_tx_power(
-        tx_power=max_tx_power,
-        net_gain_dbi=net_gain_dbi,
-        np_dbm=np_dbm,
-        mcs_snr_mbps_map=mcs_snr_mbps_map,
-    )
-    # Adjust Tx power considering Tx power backoff
-    _, tx_power = adjust_tx_power_with_backoff(
-        mcs_level=link_budget.mcs_level,
+
+    rsl_dbm = get_fspl_based_rsl(max_tx_power, net_gain_dbi)
+    snr_dbm = get_snr(rsl_dbm, np_dbm)
+    mcs_level = get_mcs_from_snr(snr_dbm, mcs_snr_mbps_map)
+
+    # Adjust tx power considering tx power backoff, i.e., tx power for computed
+    # MCS level might exceed allowed tx power
+    mcs_level, _ = adjust_tx_power_with_backoff(
+        mcs_level=mcs_level,
         mcs_snr_mbps_map=mcs_snr_mbps_map,
         min_tx_power=tx_sector_params.minimum_tx_power,
         max_tx_power=max_tx_power,
         net_gain_dbi=net_gain_dbi,
         np_dbm=np_dbm,
     )
-    # Recalculate measurements with adjusted Tx power
-    return get_measurements_from_tx_power(
+    capacity = get_link_capacity_from_mcs(mcs_level, mcs_snr_mbps_map)
+
+    # Set tx power to maximum tx power allowed for that MCS level
+    tx_backoff = get_backoff_from_mcs(mcs_level, mcs_snr_mbps_map)
```

Comment:
This change sets the tx power to the maximum allowed by the MCS class rather than minimum needed to support the class.